server/pkg/handler/chat.go

```diff
@@ -1,56 +1,51 @@
 package handler
 
 import (
-    "encoding/json"
-    "io"
-    "net/http"
-    "bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
 )
 
 type ChatRequest struct {
-    Prompt string `json:"prompt"`
+	Prompt string `json:"prompt"`
 }
 
 type ChatResponse struct {
-    Response string `json:"response"`
+	Response string `json:"response"`
 }
 
 func ChatHandler(w http.ResponseWriter, r *http.Request) {
-    var req ChatRequest
-    if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-        http.Error(w, err.Error(), http.StatusBadRequest)
-        return
-    }
-
-    response, err := generateResponse(req.Prompt)
-    if err != nil {
-        http.Error(w, err.Error(), http.StatusInternalServerError)
-        return
-    }
-
-    res := ChatResponse{Response: response}
-    json.NewEncoder(w).Encode(res)
+	var req ChatRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	response, err := generateResponse(req.Prompt)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	res := ChatResponse{Response: response}
+	json.NewEncoder(w).Encode(res)
 }
 
 func generateResponse(prompt string) (string, error) {
-    url := "http://ollama:11434/generate"  // Updated to use the service name defined in docker-compose
-    payload := map[string]string{"prompt": prompt}
-    payloadBytes, err := json.Marshal(payload)
-    if err != nil {
-        return "", err
-    }
-    resp, err := http.Post(url, "application/json", bytes.NewBuffer(payloadBytes))
-    if err != nil {
-        return "", err
-    }
-    defer resp.Body.Close()
-    body, err := io.ReadAll(resp.Body)
-    if err != nil {
-        return "", err
-    }
-    var result map[string]string
-    if err := json.Unmarshal(body, &result); err != nil {
-        return "", err
-    }
-    return result["response"], nil
+	url := "http://ollama:11434/api/generate"
+	payload := fmt.Sprintf(`{"prompt": "%s", "model": "llama3", "stream": false}`, prompt)
+	resp, err := http.Post(url, "application/json", strings.NewReader(payload))
```

Comment:
Prefer Buffer Reading with Parsing as in original. Maps are more preferred here. Can you revert it back while making your changes? Btw, the backend is completely configured by me. I still haven't pushed the changes. That's the only reason I'm requesting these changes.

Reply:
I originally used this because string-string map wouldnt have owkred because of the boolean value, I'll convert it to a map

Reply:
Kindly see the latest change, if that works